src/Assets/LightSkyDefense/Scripts/TowerAI.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Assets
+{
+    public class TowerAI : MonoBehaviour
+    {
+        public Rigidbody Projectile;
+        public Transform ProjectileSpawn;
+        public Transform Target = null;
+        private IEnumerator _coroutine;
+        public float ReloadTime = 2;
+        public float BulletSpeed = 5;
+
+        void OnTriggerEnter(Collider target)
+        {
+            Enemy enemyScript = target.gameObject.GetComponent<Enemy>();
+            if (enemyScript == null) return;
+            Target = target.transform;
+        }
+
+        void OnTriggerExit(Collider target)
+        {
+            Enemy enemyScript = target.gameObject.GetComponent<Enemy>();
+            if (enemyScript == null) return;
+            Target = null;
+        }
+
+
+        // Use this for initialization
+        void Start()
+        {
+            _coroutine = Reload(ReloadTime);
+            StartCoroutine(_coroutine);
```

Comment:
I think we need to stop the coroutine when the gameobject instance is destoyed 🤔.